libr/anal/esil.c

```diff
@@ -1904,6 +1793,8 @@ static int esil_mem_modeq_n(RAnalEsil *esil, int bits) {
 			src1 = r_anal_esil_pop (esil);
 			if (src1 && r_anal_esil_get_parm (esil, src1, &d)) {
 				r_anal_esil_pushnum (esil, d % s);
+				d = d % s;
```

Comment:
I didn't make any changes here. I merely reverted the commit. I'll look into it though. Maybe I'll make those into separate commits.